symengine/lib/symengine_wrapper.pyx

```diff
@@ -2977,11 +3029,16 @@ zoo = c2py(symengine.ComplexInf)
 nan = c2py(symengine.Nan)
 true = c2py(symengine.boolTrue)
 false = c2py(symengine.boolFalse)
+try:
+    import sympy
```

Comment:
SymPy developers are of the opinion that only imports should change and `USE_SYMENGINE=1` should be used in `sympy.core.backend` only.
For this case though, I think you'd have to write a a new class `NumberSymbol` that is a child class of `Constant` and do something like what `NumberWrapper` does.